### PR TITLE
Remove static inline to fix the clang build and added clang travis build

### PR DESCRIPTION
**Pull request by cdecker**
Clang doesn't like static inlines that are unused, so removing these fixes the clang build.

### CODE REVIEW

File: gossipd/handshake.c [hunk 1]
```diff
@@ -49,19 +49,6 @@ struct act_one {
...
-	 * : `1 byte` for the handshake version, `33 bytes` for the compressed
-	 * ephemeral public key of the initiator, and `16 bytes` for the
-	 * `poly1305` tag.
-	 */
-	BUILD_ASSERT(sizeof(act1->v) == 1);
-	BUILD_ASSERT(sizeof(act1->pubkey) == 33);
-	BUILD_ASSERT(sizeof(act1->tag) == 16);
-}
-
 /* BOLT #8:
```
Comment (rustyrussell) [inline gossipd/handshake.c]: See comments on #426 

We can't remove this.  We can, however, call it, if that makes things better.

Comment (cdecker) [reply gossipd/handshake.c]: Yep, calling them is best I think, just to avoid future confusion.

**Comment by cdecker** [comment]
Ok, added calls in the responder functions so they are no longer unused. These are `BUILD_ASSERT`s so maybe we can move them somewhere else? Is the instance of the specific type only needed to be able to call `sizeof` on a member?

**Comment by rustyrussell** [comment]
They're logically associated with the data structures, so having them as separate functions next to those makes sense.  The caller locations are less important, I suggest.

**Pull request merged**